pkgs/by-name/ap/apkeditor/package.nix

```diff
@@ -0,0 +1,164 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  jre,
+  gradle_6,
+  makeWrapper,
+}:
+
+let
+  gradle = gradle_6;
+  REAndroidLibrary =
+    args:
+    let
+      inherit (args) pname version projectName;
+      outJar = "share/${projectName}/${projectName}.jar";
+      self = stdenv.mkDerivation (
+        {
+          __darwinAllowLocalNetworking = true;
+
+          buildInputs = [ jre ];
+          nativeBuildInputs = [ gradle ];
+
+          gradleFlags = [ "-Dfile.encoding=utf-8" ];
+          gradleBuildTask = "jar";
+          doCheck = true;
+
+          inherit outJar;
+          installPhase = ''
+            runHook preInstall
+            install -Dm644 build/libs/*.jar $out/${outJar}
+            runHook postInstall
+          '';
+        }
+        // args
+        // {
+          meta.sourceProvenance = with lib.sourceTypes; [
+            fromSource
+            binaryBytecode # mitm cache
+          ];
+        }
```

Comment:
This will break when passing `meta` because the update operator isn't recursive.